### PR TITLE
Add no commit to main rule

### PR DESCRIPTION
**Pull request by hnyls2002**
## Checklist

- [ ] Format your code according to the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/en/contributor_guide.md).
- [ ] Add unit tests as outlined in the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/en/contributor_guide.md).
- [ ] Update documentation as needed, including docstrings or example tutorials.

### CODE REVIEW

**Review by zhyncs — COMMENTED** [review]
What’s the motivation?

**Comment by hnyls2002** [comment]
> What’s the motivation?

To prevent accidentally push to main

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -7,3 +10,8 @@ repos:
     rev: 24.4.2
     hooks:
       - id: black
+
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.6.0
+    hooks:
+      - id: no-commit-to-branch
```
Comment (zhyncs) [inline .pre-commit-config.yaml:17]: Shall we specify the branch args?

Comment (hnyls2002) [reply .pre-commit-config.yaml:17]: The default branch args is main and master

**Review by zhyncs — APPROVED** [review]

**Pull request merged**